huaweicloud/services/apig/resource_huaweicloud_apig_acl_policy_associate.go

```diff
@@ -0,0 +1,230 @@
+		DeleteContext: resourceAclPolicyAssociateDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAclPolicyAssociateImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The region where the ACL policy and the APIs are located.",
+			},
+			"instance_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the dedicated instance to which the APIs and the ACL policy belong.",
+			},
+			"policy_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ACL Policy ID for APIs binding.",
+			},
+			"publish_ids": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Description: "The publish IDs corresponding to the APIs bound by the ACL policy.",
+			},
+		},
+	}
+}
+
+func resourceAclPolicyAssociateCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.ApigV2Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+
+	var (
+		instanceId = d.Get("instance_id").(string)
+		policyId   = d.Get("policy_id").(string)
+		publishIds = d.Get("publish_ids").(*schema.Set)
+
+		opts = acls.BindOpts{
+			InstanceId: instanceId,
+			PolicyId:   policyId,
+			PublishIds: utils.ExpandToStringListBySet(publishIds),
+		}
+	)
+	_, err = acls.Bind(client, opts)
+	if err != nil {
+		return diag.Errorf("error binding policy to the API: %s", err)
+	}
+	d.SetId(fmt.Sprintf("%s/%s", instanceId, policyId))
+
+	return resourceAclPolicyAssociateRead(ctx, d, meta)
+}
+
+func buildAclPolicyListOpts(instanceId, policyId string) acls.ListBindOpts {
+	return acls.ListBindOpts{
+		InstanceId: instanceId,
+		PolicyId:   policyId,
+		Limit:      500, // This limitation can be removed after the parameter 'offset' is fixed.
+	}
+}
+
+func flattenApiPublishIdsForAclPolicy(apiList []acls.AclBindApiInfo) []string {
+	if len(apiList) < 1 {
+		return nil
+	}
+
+	result := make([]string, len(apiList))
+	for i, val := range apiList {
+		result[i] = val.PublishId
+	}
+	return result
+}
+
+func resourceAclPolicyAssociateRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.ApigV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating APIG v2 client: %s", err)
+	}
+
+	var (
+		instanceId = d.Get("instance_id").(string)
+		policyId   = d.Get("policy_id").(string)
+		opts       = buildAclPolicyListOpts(instanceId, policyId)
+	)
+
+	resp, err := acls.ListBind(client, opts)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error getting API information from server")
```

Comment:
looks the error message not very fit here

Reply:
done.